services/diversity/src/services/richness.js

```diff
@@ -12,105 +12,109 @@ module.exports = (RichnessPersistence, restAPI) => {
      *
      * @returns {Object[]} Number of inferred and observed species for the desired group.
      */
-    getNumberSpecies: async (areaType, areaId, group = 'all') => {
-      const data = [
-        {
-          id: 'total',
-          inferred: 40,
-          observed: 4,
-          region_observed: 110,
-          region_inferred: 130,
-        },
-        {
-          id: 'endemic',
-          inferred: 20,
-          observed: 25,
-          region_observed: 120,
-          region_inferred: 100,
-        },
-        {
-          id: 'invasive',
-          inferred: 20,
-          observed: 10,
-          region_observed: 90,
-          region_inferred: 80,
-        },
-        {
-          id: 'threatened',
-          inferred: 15,
-          observed: 20,
-          region_observed: 90,
-          region_inferred: 100,
-        },
-      ];
+    getNumberOfSpecies: async (areaType, areaId, group = 'all') => {
+      const promises = [];
+      let result;
       switch (group) {
         case 'total':
-          return [data[0]];
+          promises.unshift(RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId));
+          break;
         case 'endemic':
-          return [data[1]];
+          promises.unshift(RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId));
+          break;
         case 'invasive':
-          return [data[2]];
+          promises.unshift(RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId));
+          break;
         case 'threatened':
-          return [data[3]];
+          promises.unshift(RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId));
+          break;
         case 'all':
+          promises.unshift(
+            RichnessPersistence.findTotalNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findEndemicNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findInvasiveNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findThreatenedNumberOfSpecies(areaType, areaId),
+          );
+          break;
         default:
-          return data;
+          return [];
       }
+
+      if (!promises) {
+        throw new Error('There\'s not any values of number of species');
+      } else if (group !== 'all') {
+        result = await Promise.all(promises)
+          .then(response => response.map(item => ({ id: group, ...item[0] })))
+          .catch();
+      } else {
+        const ids = ['total', 'endemic', 'invasive', 'threatened'];
+        result = await Promise.all(promises)
+          .then(response => response.map((item, i) => ({ id: ids[i], ...item[0] })))
+          .catch();
+      }
+      return result;
     },
 
     /**
-     * Get thresholds for the number of species for the given area type in the given group
+     * Get thresholds for the number of species for the given area type and area id
```

Comment:
Yo don't get the thresholds for the areaId, but for the region to which the areaId belongs